detectors/node/opentelemetry-resource-detector-aws/src/detectors/AwsEcsDetector.ts

```diff
@@ -77,13 +91,137 @@ export class AwsEcsDetector implements Detector {
+      return new Resource({
+        [SemanticResourceAttributes.CONTAINER_NAME]: hostName || '',
+        [SemanticResourceAttributes.CONTAINER_ID]: containerId || '',
+      });
+    }
+
+    return Resource.empty();
+  }
+
+  private async _getMetadataV4Resource(): Promise<Resource> {
+    const metadataUrl = getEnv().ECS_CONTAINER_METADATA_URI_V4;
+
+    if (!metadataUrl) {
+      return Resource.empty();
+    }
+
+    const [responseContainer, responseTask] = await Promise.all([
+      AwsEcsDetector._getUrlAsJson(metadataUrl),
+      AwsEcsDetector._getUrlAsJson(`${metadataUrl}/task`),
+    ]);
+
+    const launchType: string = responseTask['LaunchType'];
+    const taskArn: string = responseTask['TaskARN'];
+
+    const baseArn: string = taskArn.substring(0, taskArn.lastIndexOf(':'));
+    const cluster: string = responseTask['Cluster'];
+
+    const clusterArn =
+      cluster.indexOf('arn:') == 0 ? cluster : `${baseArn}:cluster/${cluster}`;
+
+    const containerArn: string = responseContainer['ContainerARN'];
+
+    // https://github.com/open-telemetry/opentelemetry-specification/blob/main/semantic_conventions/resource/cloud_provider/aws/ecs.yaml
+    return new Resource({
+      [SemanticResourceAttributes.AWS_ECS_CONTAINER_ARN]: containerArn,
+      [SemanticResourceAttributes.AWS_ECS_CLUSTER_ARN]: clusterArn,
+      [SemanticResourceAttributes.AWS_ECS_LAUNCHTYPE]:
+        launchType?.toLowerCase(),
+      [SemanticResourceAttributes.AWS_ECS_TASK_ARN]: taskArn,
+      [SemanticResourceAttributes.AWS_ECS_TASK_FAMILY]: responseTask['Family'],
+      [SemanticResourceAttributes.AWS_ECS_TASK_REVISION]:
+        responseTask['Revision'],
+    }).merge(AwsEcsDetector._getLogResource(responseContainer));
+  }
+
+  private static _getLogResource(containerMetadata: any): Resource {
+    if (
+      containerMetadata['LogDriver'] != 'awslogs' ||
+      !containerMetadata['LogOptions']
+    ) {
+      return Resource.EMPTY;
     }
-    return undefined;
+
+    const containerArn = containerMetadata['ContainerARN']!;
+    const logOptions = containerMetadata['LogOptions'] as AwsLogOptions;
+
+    const logsRegion =
+      logOptions['awslogs-region'] ||
+      AwsEcsDetector._getRegionFromArn(containerArn);
+
+    const awsAccount = AwsEcsDetector._getAccountFromArn(containerArn);
+
+    const logsGroupName = logOptions['awslogs-group']!;
+    const logsGroupArn = `arn:aws:logs:${logsRegion}:${awsAccount}:log-group:${logsGroupName}`;
+    const logsStreamName = logOptions['awslogs-stream']!;
+    const logsStreamArn = `arn:aws:logs:${logsRegion}:${awsAccount}:log-group:${logsGroupName}:log-stream:${logsStreamName}`;
+
+    return new Resource({
+      [SemanticResourceAttributes.AWS_LOG_GROUP_NAMES]: [logsGroupName],
+      [SemanticResourceAttributes.AWS_LOG_GROUP_ARNS]: [logsGroupArn],
+      [SemanticResourceAttributes.AWS_LOG_STREAM_NAMES]: [logsStreamName],
+      [SemanticResourceAttributes.AWS_LOG_STREAM_ARNS]: [logsStreamArn],
```

Comment:
I am not very familiar with these properties, but I see they are being set as arrays with one value. Are they always expected to have just one item? can there be more?

Reply:
It could be more if the ECS task had multiple containers, and the tracer was reporting about all of them. This is more the kind of thing that the OpenTelemetry Collector would do though.